library/src/main/java/com/novoda/downloadmanager/Wait.java

```diff
@@ -0,0 +1,52 @@
+package com.novoda.downloadmanager;
+
+import android.support.annotation.Nullable;
+
+import com.novoda.notils.logger.simple.Log;
+
+final class Wait {
+
+    private Wait() {
+        // Uses static factory method.
+    }
+
+    static <T> ThenPerform<T> waitFor(@Nullable Object instanceToWaitFor, Object lock) {
```

Comment:
Wait is now more generic and can be used for any object in the library for flexibility. We could interface this out to enforce type safety but this would mean casting in some places where we rely on a different scope.

Reply:
yeah, I don't really like the use of all these `Object`s...
Just making a test with interfaces, is it gonna be too limiting?

Reply:
79e4494